### PR TITLE
[feat]: [PL-31352]: In List Role assignment by Scope Filters List API,  added more filters support.

### PR DESCRIPTION
**Pull request by Jimit09Gandhi**
## Describe your changes
In List Role assignment by Scope Filters List API, adding following filters.
- Harness Managed Filter 
- Disabled Filter
- Scope Filter

Test Cases:
List Role assignments on following filters and its combination:

- Principal Filter
- PrincipalType Filter
- Harness Managed Filter 
- Disabled Filter
- Scope Filter

## Checklist
- [X] I've documented the changes in the PR description.
- [X] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44375)

### CODE REVIEW

**Comment by Jimit09Gandhi** [comment]
trigger ut5

**Comment by Jimit09Gandhi** [comment]
trigger ut6

**Comment by Jimit09Gandhi** [comment]
trigger ut7

**Comment by Jimit09Gandhi** [comment]
trigger ut8

**Comment by Jimit09Gandhi** [comment]
trigger ut9

File: access-control/contracts/api/src/main/java/io/harness/accesscontrol/roleassignments/api/RoleAssignmentFilterV2.java [hunk 1]
```diff
@@ -31,4 +32,9 @@ public class RoleAssignmentFilterV2 {
   @Schema(description = "Filter role assignments based on scope filters")
   Set<ScopeSelector> scopeFilters = new HashSet<>();
   @Schema(description = "Filter role assignments based on principal") PrincipalDTO principalFilter;
+  @Schema(description = "Filter role assignments based on role assignments being harness managed")
+  Set<Boolean> harnessManagedFilter;
+  @Schema(description = "Filter role assignments based on whether they are enabled or disabled")
+  Set<Boolean> disabledFilter;
+  @Schema(description = "Filter role assignments based on principal type") Set<PrincipalType> principalTypeFilter;
```
Comment (meenakshi-harness) [inline access-control/contracts/api/src/main/java/io/harness/accesscontrol/roleassignments/api/RoleAssignmentFilterV2.java:39]: Did we try if we can use principalFilter itself for this? This looks like redundant to me having 2 filters fields for same filter type. We can create new PrincipalFilterDTO with all fields optional?

**Review by Jimit09Gandhi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44375)* status: 4 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`access-control/contracts/api/src/main/java/io/harness/accesscontrol/roleassignments/api/RoleAssignmentFilterV2.java` line 39 at r2](https://reviewable.io/reviews/harness/harness-core/44375#-NPDsiQNZ__u60Y_ze2z-r2-39:-NPTG8ce5DC00n461EdK:b-lqtpyv) ([raw file](https://github.com/harness/harness-core/blob/dbee48f2f0ff563a4f57ca3cb66e8c5884ae4a7f/access-control/contracts/api/src/main/java/io/harness/accesscontrol/roleassignments/api/RoleAssignmentFilterV2.java#L39)):*
<details><summary><i>Previously, meenakshi-harness (Meenakshi Raikwar) wrote…</i></summary><blockquote>

Did we try if we can use principalFilter itself for this? This looks like redundant to me having 2 filters fields for same filter type. We can create new PrincipalFilterDTO with all fields optional?
</blockquote></details>

As discussed, both are different filters. principalFilter for searching on specific Principal and principalTypeFilter for querying on certain types.

File: access-control/service/src/main/java/io/harness/accesscontrol/roleassignments/api/RoleAssignmentDTOMapper.java [hunk 2]
```diff
@@ -134,10 +135,11 @@ public static RoleAssignmentFilter fromV2(RoleAssignmentFilterV2 object) {
             .roleFilter(object.getRoleFilter() == null ? new HashSet<>() : object.getRoleFilter())
             .resourceGroupFilter(
                 object.getResourceGroupFilter() == null ? new HashSet<>() : object.getResourceGroupFilter())
-            .disabledFilter(object.getDisabledFilter() == null ? new HashSet<>() : object.getDisabledFilter())
+            .disabledFilter(
+                object.getDisabledFilter() == null ? new HashSet<>() : Sets.newHashSet(object.getDisabledFilter()))
             .managedFilter(Objects.isNull(object.getHarnessManagedFilter())
                     ? ManagedFilter.NO_FILTER
-                    : buildFromSet(object.getHarnessManagedFilter()))
+                    : buildFromSet(Sets.newHashSet(object.getHarnessManagedFilter())))
```
Comment (meenakshi-harness) [inline access-control/service/src/main/java/io/harness/accesscontrol/roleassignments/api/RoleAssignmentDTOMapper.java:142]: Why do we need this set conversion now?

**Review by ashishsanodia-harness — APPROVED** [review]
> Reviewed 1 of 2 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44375)* status: 4 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by Jimit09Gandhi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44375)* status: 4 file(s) to review. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`access-control/service/src/main/java/io/harness/accesscontrol/roleassignments/api/RoleAssignmentDTOMapper.java` line 142 at r3](https://reviewable.io/reviews/harness/harness-core/44375#-NPDsiQNZ__u60Y_ze3--r3-142:-NPW64G20cBTO6Xtr0m_:b-h6trh1) ([raw file](https://github.com/harness/harness-core/blob/537571b11df08c00c363126e9c90bc85ab67d436/access-control/service/src/main/java/io/harness/accesscontrol/roleassignments/api/RoleAssignmentDTOMapper.java#L142)):*
<details><summary><i>Previously, meenakshi-harness (Meenakshi Raikwar) wrote…</i></summary><blockquote>

Why do we need this set conversion now? 
</blockquote></details>

Changed. done.

**Review by meenakshi-harness — APPROVED** [review]
> Reviewed 1 of 2 files at r2, 1 of 3 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44375)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Review by ashishsanodia-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44375)* status: :shipit: complete! ✓

**Review by pbarapatre10 — APPROVED** [review]
> Reviewed 1 of 2 files at r2, 1 of 3 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44375)* status: :shipit: complete! ✓

**Review by pbarapatre10 — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44375#-:-NPWixS-3HK0Nm55VOvI:bnfp4nl)* status: :shipit: complete! ✓

**Review by ashishsanodia-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44375)* status: :shipit: complete! ✓

**Review by meenakshi-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44375)* status: :shipit: complete! ✓

**Review by manish-manchanda — APPROVED** [review]
> Reviewed 1 of 2 files at r2, 1 of 3 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44375)* status: :shipit: complete! ✓

**Pull request merged**